corehq/db/config.yml

```diff
@@ -0,0 +1,20 @@
+# HQ database configuration for multiple database. This gets mapped to Django's DATABASES dict
```

Comment:
I think splitting settings.py up into multiple files might be reasonable. Where would those other files be located? Would settings.py become a settings package?

@snopoke wrote:
> I'm not convinced about the SO solution you linked to. I think that works for basic key value settings but not for something like this which is more complex and going to look very different on different environments.

I think that was the whole point: complex settings can be processed by (and not intermingled with) sensible python code that knows how to deal with complex (or simple) data structures. That code should be well organized and able to be refactored like any other part of the application, not thrown into this huge flat interdependent mess of globals like most Django settings.py files eventually become.

The part that really resonated with me was this:

> over time enough application programming creeps in that it gets hard to handle; primarily as settings become derivative and/or codependent. There can be good justifications for settings to react according to the local settings which forces the import of a local_settings file to creep up toward the middle of settings.py. I find things start to get messy as that happens.

I think we have that problem big time, and I haven't come across a better strategy to deal with it.